src/mono/wasm/runtime/exports.ts

```diff
@@ -70,6 +70,8 @@ import { EmscriptenModule } from "./types/emscripten";
 import { mono_run_main, mono_run_main_and_exit } from "./run";
 import { diagnostics } from "./diagnostics";
 import { dotnet_browser_can_use_simple_digest_hash, dotnet_browser_simple_digest_hash } from "./crypto-worker";
+import { mono_wasm_pthread_on_pthread_created } from "./pthreads/worker";
```

Comment:
This would get included even in non-threaded runtime. Maybe we could do something with `rollup` to link it out. See `rollup.config.js` -> `productVersion` and also `rollup-plugin-consts`, I think it could eliminate dead code on constants.

Reply:
I think I can almost make this work, but how do I make one of these rollup constants visible in `dotnet.cjs.lib.js`? I need to conditionally fill `linked_functions`.

Reply:
Create `dotnet.cjs.mt.lib.js` and do it in build files ?

We need to do something smarter about those `linked_functions` in the future. 
I'm not confident what we have is the best solution.
There is some crazy transormation on it by emscripten.

Reply:
oh, I guess the lib.js files go through emcc's weird C preprocessor JS thing, so I can do something with `#ifdef`.

Reply:
oh god, it's not the real C preprocessor. it's their own homegrown thing...

https://github.com/emscripten-core/emscripten/blob/54d1aa0917e91c60533dd0b6bee2212d487cc78a/src/parseTools.js#L37

Reply:
Ok, pushed a fix for this.  verified that a single-threaded Release build of `dotnet.js` doesn't have any of the pthread support code in it